presto-main/src/test/java/com/facebook/presto/operator/scalar/TestZipWithFunction.java

```diff
@@ -0,0 +1,74 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.RowType;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static java.util.Arrays.asList;
+
+public class TestZipWithFunction
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testSameLength()
+            throws Exception
+    {
+        assertFunction("zip_with(ARRAY[1, 2], ARRAY['a', 'b'], (x, y) -> (y, x))",
+                new ArrayType(new RowType(ImmutableList.of(createVarcharType(1), INTEGER), Optional.empty())),
+                ImmutableList.of(ImmutableList.of("a", 1), ImmutableList.of("b", 2)));
+
+        assertFunction("zip_with(ARRAY[1, 2], ARRAY[CAST('a' AS VARCHAR), CAST('b' AS VARCHAR)], (x, y) -> (y, x))",
+                new ArrayType(new RowType(ImmutableList.of(VARCHAR, INTEGER), Optional.empty())),
+                ImmutableList.of(ImmutableList.of("a", 1), ImmutableList.of("b", 2)));
+
+        assertFunction("zip_with(ARRAY[], ARRAY[], (x, y) -> (y, x))",
+                new ArrayType(new RowType(ImmutableList.of(UNKNOWN, UNKNOWN), Optional.empty())),
+                ImmutableList.of());
+
+        assertFunction("zip_with(ARRAY[NULL], ARRAY[NULL], (x, y) -> (y, x))",
+                new ArrayType(new RowType(ImmutableList.of(UNKNOWN, UNKNOWN), Optional.empty())),
+                ImmutableList.of(asList(null, null)));
+
+        assertFunction("zip_with(ARRAY[1, 1], ARRAY[1, 2], (x, y) -> x + y)",
+                new ArrayType(INTEGER),
+                ImmutableList.of(2, 3));
+    }
+
+    @Test
+    public void testDifferentLength()
+            throws Exception
+    {
+        assertInvalidFunction("zip_with(ARRAY[1], ARRAY['a', 'b'], (x, y) -> (y, x))", "Two arrays have different length");
+        assertInvalidFunction("zip_with(ARRAY[NULL, 2], ARRAY['a'], (x, y) -> (y, x))", "Two arrays have different length");
+        assertInvalidFunction("zip_with(ARRAY[1, NULL], ARRAY[3, 2, 1], (x, y) -> x + y)", "Two arrays have different length");
+    }
+
+    @Test
+    public void testWithNull()
+            throws Exception
+    {
+        assertFunction("zip_with(CAST(NULL AS ARRAY(UNKNOWN)), ARRAY[], (x, y) -> (y, x))",
```

Comment:
Please test cases where
* the array contains null, but the lambda returns non-null always
* the array contains null, the lambda returns null when one of the input is null
* the array doesn't contain null, the lambda returns null for some non-null input